packages/web3-eth-contract/src/types.ts

```diff
@@ -281,7 +281,7 @@ export interface NonPayableMethodObject<Inputs = unknown[], Outputs = unknown[]>
 	 * @returns - The gas amount estimated.
 	 */
 	estimateGas<ReturnFormat extends DataFormat = typeof DEFAULT_RETURN_FORMAT>(
-		options?: NonPayableCallOptions,
+		options?: Omit<NonPayableCallOptions, 'nonce'>,
```

Comment:
Looks like `nonce` is allowed for `estimateGas`
<img width="410" alt="image" src="https://user-images.githubusercontent.com/20375610/233437916-767633de-3c6d-4a13-a45e-78f0ba4abb92.png">